flake.nix

```diff
@@ -0,0 +1,31 @@
+{
+  description = "Hardware and Firmware of the Solo Quadruped Master Board.";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs?ref=nixos-unstable";
+  };
+
+  outputs =
+    { self, nixpkgs }:
+    let
+      pkgs = nixpkgs.legacyPackages.x86_64-linux;
+    in
+    {
+      packages.x86_64-linux.sdk = pkgs.stdenv.mkDerivation {
+
+        pname = "odri-masterboard-sdk";
+        version = "nightly";
```

Comment:
that can be 1.0.7

Reply:
since it's built from source i'd assume the version would be set when building a release through CI, but we can just stick the latest version here too of course

Reply:
the release process will have to bump the number here too.
We could also extract that from eg. `sdk/master_board_sdk/package.xml` instead.
The idea is that a downstream package could check `odri-masterboard-sdk.version` and have a meaningful info

Reply:
unfortunately, xml parsing seems kinda hard to do within nix expressions, see https://github.com/NixOS/nixpkgs/issues/57239

we might have a way to do it via https://github.com/kanwren/nix-parsec/pull/6, but the PR has not been merged and was open in 2022...

see also https://nixos.wiki/wiki/String-parsing_in_Nix

i guess another way to have a ssot for the version is to... keep a JSON, and generate both package.xml and use fromJSON in the nix flake, but it's kinda ugly?

Reply:
we _could_ also [parse XML using regex](https://stackoverflow.com/a/1732454) since the thing we need to get out is kinda fixed and pretty simple (maybe sth like `<version>\s*([\d.]+)\s*</version>` ?)

Reply:
I guess we could add that in a lib somewhere, probably https://github.com/Gepetto/nix, so that we can use it in our other projects.

What do you think ?

Reply:
oh wow, yq does xml too! good to know

a shared lib would be nice yeah, ideally as a separate thing so that anyone can use it (since there's demand for it, https://github.com/NixOS/nixpkgs/issues/57239 for example)

can it be put in Gepetto/nix and still be "imported" independently from the other things residing in Gepetto/nix ?

Reply:
Good idea :)

Maybe in that case it would be better to create a new repo, with only that.

While here I would like to have the same feature, but for `pyproject.toml`

Any idea for a name ?

Reply:
as far as pyproject.toml goes, fromTOML already exists i think: https://github.com/mozilla/nixpkgs-mozilla/blob/master/lib/parseTOML.nix